client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -82,30 +86,24 @@ const CheckoutThankYou = React.createClass( {
 		window.scrollTo( 0, 0 );
 	},
 
-	componentWillReceiveProps() {
+	componentWillReceiveProps( nextProps ) {
 		this.redirectIfThemePurchased();
-		this.refreshSitesAndSitePlansIfPlanPurchased();
-	},
-
-	hasPlanOrDomainRegistration() {
-		return getPurchases( this.props ).some( purchase => isPlan( purchase ) || isDomainRegistration( purchase ) );
-	},
 
-	refreshSitesAndSitePlansIfPlanPurchased() {
-		if ( this.props.receipt.hasLoadedFromServer && this.hasPlanOrDomainRegistration() ) {
-			// Refresh selected site plans if the user just purchased a plan
+		if ( ! this.props.receipt.hasLoadedFromServer && nextProps.receipt.hasLoadedFromServer && this.hasPlanOrDomainRegistration( nextProps ) ) {
 			this.props.refreshSitePlans( this.props.selectedSite.ID );
-
-			// Refresh the list of sites to update the `site.plan` property
-			// needed to display the plan name on the right of the `Plans` menu item
-			Dispatcher.handleViewAction( {
```

Comment:
This isn't needed anymore, because we update `sites-list` from the checkout action [here](https://github.com/Automattic/wp-calypso/blob/master/client/lib/sites-list/index.js#L34).